libweb3jsonrpc/Debug.cpp

```diff
@@ -134,3 +137,34 @@ Json::Value Debug::debug_storageAt(string const& _blockHashOrNumber, int _txInde
 		ret[toHex(toCompactBigEndian(i.first, 1), 2, HexPrefix::Add)] = toHex(toCompactBigEndian(i.second, 1), 2, HexPrefix::Add);
 	return ret;
 }
+
+Json::Value Debug::debug_traceCall(Json::Value const& _call, std::string const& _blockNumber, Json::Value const& _options)
+{
+	Json::Value ret;
+	try
+	{
+		Block temp = m_eth.asOf(jsToBlockNumber(_blockNumber));
+		TransactionSkeleton ts = toTransactionSkeleton(_call);
+		if (!ts.from) {
+			ts.from = m_ethAccounts.defaultTransactAccount();
+		}
+		u256 nonce = temp.transactionsFrom(ts.from);
+		u256 gas = ts.gas == Invalid256 ? m_eth.gasLimitRemaining() : ts.gas;
+		u256 gasPrice = ts.gasPrice == Invalid256 ? m_eth.gasBidPrice() : ts.gasPrice;
+		Transaction t(ts.value, gasPrice, gas, ts.to, ts.data, nonce);
```

Comment:
I thought about that but I was not to sure if we should do that.
Depends how this function is used: there s the use case where one want to execute a transaction and verify if everything goes well prior to broadcast it. in this case maybe we should not force funding cause then broadcasting it might lead to a different behavior.
btw this actually also applies for a call. if we force funding it might be that the result of the call will not reflect the reality of the current user balance.